cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/DynamicDatasetCache.java

```diff
@@ -129,6 +124,11 @@ public DynamicDatasetCache(SystemDatasetInstantiator instantiator,
     }
   }
 
+  @Override
+  public void releaseDataset(Dataset dataset) {
+    dismissDataset(dataset);
```

Comment:
yes. Since the javadoc states: it is up to the implementation, I started with the simplest possible one. Another option would be to remove releaseDataset() for now. Only that that will break compatibility again when we introduce it.